models/ios/ntp_global/ios_ntp_global.yaml

```diff
@@ -0,0 +1,228 @@
+module: ios_ntp_global
+short_description: ntp_global resource module
+description:
+- This module provides declarative management of ntp on Cisco IOS devices.
+version_added: 2.5.0
+author:
+- Sagar Paul (@KB-perByte)
+notes:
+- Tested against Cisco IOSv Version 15.6.
+- This module works with connection C(network_cli).
+options:
+  config:
+    description: A dictionary of ntp options
+    elements: dict
+    suboptions:
+      access_group:
+        description: Control NTP access
+        type: dict
+        suboptions:
+          ipv4: 
+            description:  ipv4 access lists (Default not idempotent)
+            type: bool
+          ipv6: 
+            description:  ipv6 access lists (Default not idempotent)
+            type: bool
+          peer: 
+            description: Provide full access
+            type: dict
+            suboptions:
+              access_list: &access_group_acl
+                description:  name or number of access list
+                type: str
+              kod: &kod
+                description: Send a Kiss-o-Death packet for failing peers
+                type: bool
+          query-only: 
+            description:  Allow only control queries
+            type: dict
+            suboptions: 
+              access_list: *access_group_acl
+              kod: *kod
+          serve: 
+            description:  Provide server and query access
+            type: dict
+            suboptions:
+              access_list: *access_group_acl
+              kod: *kod
+          serve-only: 
+            description:  Provide only server access
+            type: dict
+            suboptions:
+              access_list: *access_group_acl
+              kod: *kod
+      allow:
+        description: Allow processing of packets
+        type: dict
+        elements: dict
+        suboptions:
+          control:
+            description: Allow processing control mode packets
+            type: bool
+          private:
+            description: Allow processing private mode packets
+            type: bool
+          rate_limit_delay:
+            description: Rate limiting delays
+            type: int
+      authenticate:
+        description: Authenticate time sources
+        type: bool
+      authentication_key:
+        description: Authentication key for trusted time sources
+        type: dict
+        elements: dict
+        suboptions:
+          key_num:
+            description: Key number
+            type: int
+          authentication_type:
+            description: Authentication type
+            type: str
+          password:
+            description: Password
+            type: str
```

Comment:
This should have `no_log: True`.

Reply:
I don't have this option on the device! which device are you referring to?

```
csr1000v-1(config)#ntp authentication-key ?
  <1-4294967295>  Key number

csr1000v-1(config)#ntp authentication-key 2 ?
  cmac-aes-128   CMAC-AES-128 (digest length = 128 bits, key length = [16 or 32] bytes)
  hmac-sha1      HMAC-SHA1 (digest length = 160 bits, key length = [1-32] bytes)
  hmac-sha2-256  HMAC-SHA2-256 (digest length = 256 bits, key length = [1-32] bytes)
  md5            MD5 authentication

csr1000v-1(config)#ntp authentication-key 2 md5 ?
  WORD  Authentication key

csr1000v-1(config)#ntp authentication-key 2 md5 abs ?
  <0-4294967295>  Authentication key encryption type
  <cr>            <cr>
```

Reply:
mybad! got it

Reply:
If `authentication_type` refers to cmac, md5, etc., can we rename it to `algorithm` like it is done in the EOS model?

Reply:
makes sense! Thanks